spec/features/consumer/shopping/embedded_shopfronts_spec.rb

```diff
@@ -94,6 +94,7 @@
         wait_for_shop_loaded
         find('ul.right li#login-link a').click
         login_with_modal
+        expect(page).to_not have_selector 'div.login-modal', visible: true
```

Comment:
This is the first of three waiting statements. The last one should be enough if it was working. I guess that these three statements just slow down execution and make the failure less likely. We need to find a way to determine when all the JS work is done.